stash/src/controller/mgxAirdropController.ts

```diff
@@ -0,0 +1,173 @@
+
+const SNAPSHOTS = [FIRST_SNAPSHOT_BLOCK_HASH, SECOND_SNAPSHOT_BLOCK_HASH]
+
+export async function checkEligibility(req: Request, res: Response) {
+  /*
+    #swagger.tags = ['mgx-airdrop']
+    #swagger.summary = 'Check eligibility for MGX airdrop'
+    #swagger.description = 'Verifies if the provided address is eligible for the airdrop based on snapshots.'
+    #swagger.responses[200] = {
+      description: 'Successful response',
+      schema: {
+        mangataXAddress: '5Gxkb...PDnnR',
+        isEligible: true
+      }
+    }
+    #swagger.responses[400] = {
+      description: 'Validation error'
+    }
+    #swagger.responses[500] = {
+      description: 'Internal Server Error'
+    }
+  */
+  try {
+    const { address } = req.params
+
+    mgxAirdropEligibilitySchema.validateSync({ address })
+
+    const snapshotPromises = SNAPSHOTS.map((snapshot) =>
+      getEligibilityAtBlockN(snapshot, address)
+    )
+
+    const snapshots = await Promise.all(snapshotPromises)
+    const weight = snapshots.reduce(
+      (acc, snapshot) => acc.add(snapshot.weight),
+      BN_ZERO
+    )
+
+    return res.json({
+      mangataXAddress: address,
+      isEligible: weight.gtn(0),
+    })
+  } catch (e) {
+    await errorHandler.handle(res, e)
+  }
+}
+
+export async function linkAddress(req: Request, res: Response) {
+  /*
+    #swagger.tags = ['mgx-airdrop']
+    #swagger.summary = 'Link Ethereum address to MangataX address'
+    #swagger.description = 'Links the provided Ethereum address to the provided MangataX address.'
+    #swagger.parameters['body'] = {
+      in: 'body',
+      description: 'JSON object containing the Ethereum and MangataX addresses to be linked',
+      required: true,
+      schema: {
+        ethereumAddress: '0x...',
+        mangataXAddress: '5Gxkb...PDnnR'
+      }
+    }
+    #swagger.responses[200] = {
+      description: 'Successful response',
+      schema: {
+        ethereumAddress: '0x...',
+        mangataXAddress: '5Gxkb...PDnnR'
+      }
+    }
+    #swagger.responses[400] = {
+      description: 'Validation error'
+    }
+    #swagger.responses[500] = {
+      description: 'Internal Server Error'
+    }
+  */
+  try {
+    const { mangataXAddress, ethereumAddress } = req.body
+
+    mgxAirdropLinkAddressSchema.validateSync({
+      mangataXAddress,
+      ethereumAddress,
+    })
+
+    const { signature } = mgxAirdropSignatureSchema.validateSync({
+      signature: req.headers['x-signature'],
+    })
+
+    const isSignatureValid = verifySignature(
+      { mangataXAddress, ethereumAddress },
+      signature
+    )
+
+    if (!isSignatureValid) {
+      throw new BadRequestException('Invalid signature')
+    }
+
+    const record = await mgxAirdropRepository
+      .search()
+      .where('mangataXAddress')
+      .eq(mangataXAddress)
+      .returnFirst()
+
+    if (record) {
+      record.ethereumAddress = ethereumAddress
+      await mgxAirdropRepository.save(record)
+
+      return res.json({
```

Comment:
what do you return if there is no record found?

Reply:
The function just continues and creates one